Sources/Apodini/Information/Authorization.swift

```diff
@@ -0,0 +1,123 @@
+                  let base64data = Data(base64Encoded: credentials),
+                  let usernameAndPassword = String(data: base64data, encoding: .utf8) else {
+                return nil
+            }
+            
+            let splitString = usernameAndPassword.split(separator: ":", maxSplits: 1)
+            
+            guard splitString.count == 2 else {
+                return nil
+            }
+            
+            return (String(splitString[0]), String(splitString[1]))
+        }
+        
+        /// The brearer token the `Authorization.Value` is based on bearer authorization
+        public var bearerToken: String? {
+            guard type == "Bearer" else {
+                return nil
+            }
+            
+            return credentials
+        }
+        
+        
+        /// Crerates a new `Authorization` instance that represents a basic authorization username and password combination
+        /// - Parameters:
+        ///   - username: The username that is encoded together with the password.
+        ///   - password: The password that is encoded together with the username.
+        /// - Returns: An `Authorization` instance encoding the `username` and `password` using the basic authorization mechanism
+        /// - Warning: Please be aware  that the username and password is **not** encrrypted in a basic token, it is just base64 encoded!
+        public static func basic(username: String, password: String) -> Self {
+            let credentials = Data("\(username):\(password)".utf8).base64EncodedString()
+            return Self(type: "Basic", credentials: credentials)
+        }
+        
+        /// Crerates a new `Authorization` instance that represents a bearer authorization token
+        /// - Parameter token: The token that is encoded in the bearer authorization.
+        /// - Returns: An `Authorization` instance encoding the `token` using the bearer authorization mechanism
+        public static func bearer(_ token: String) -> Self {
+            Self(type: "Bearer", credentials: token)
+        }
+        
+        
+        /// Crerates a new `Authorization` instance
+        /// - Parameters:
+        ///   - type: The authorization type
+        ///   - credentials: The authorization credentials
+        public init(type: String, credentials: String) {
+            self.type = type
+            self.credentials = credentials
+        }
+        
+        public init?(rawValue: String) {
+            let substrings = rawValue.split(separator: " ", maxSplits: 1)
+            guard substrings.count == 2 else {
+                return nil
+            }
+            self = Self(type: String(substrings[0]), credentials: String(substrings[1]))
+        }
+    }
+    
+    
+    public static var key: String {
+        "Authorization"
+    }
+    
+    
+    public private(set) var value: Value
+    
+    
+    public var rawValue: String {
+        value.rawValue
+    }
```

Comment:
I guess I would prefer to always have these properties at the top of the struct, even if a custom Value is defined (it's clearer and more consistent with the other Information structs). But I guess that's a matter of taste. Would also apply to eg. `ETag.swift`

Reply:
I am trying to have a consistent style in the project that is as followed:
- Nested Types
- Static Properties (First Stored then Computed)
- Instance Properties (First Stored then Computed)
- Subsctipts
- Initializer
- Static Functions
- Instance Functions

But I agree that the long nested type makes it a bit hard to read. I have added it below the main type in an extension, this might make it a bit clearer how the code is structured.
In the long term we could enforce this file order rule using SwiftLint, We should probably take a look at that.